transport/internet/system_listener.go

```diff
@@ -80,6 +83,11 @@ func (dl *DefaultListener) Listen(ctx context.Context, addr net.Addr, sockopt *S
 		policyFunc := func(upstream net.Addr) (proxyproto.Policy, error) { return proxyproto.REQUIRE, nil }
 		l = &proxyproto.Listener{Listener: l, Policy: policyFunc}
 	}
+	if err == nil && ds {
+		if e := os.Chmod(address, 0666); e != nil {
```

Comment:
move this to the `ds body (if body)` above. there's no need to check nil error and create a temp bool.

Reply:
ok

Reply:
do not create unnecessary bool `ds = true`, and handle err in the domain_socket body.
```golang
... ...
		} else {
			// normal unix domain socket needs lock
			locker := &FileLocker{
				path: address + ".lock",
			}
			if err := locker.Acquire(); err != nil {
				return nil, err
			}
			if err := os.Chmod(address, 0666); err != nil {
				return nil, err
			}
			ctx = context.WithValue(ctx, address, locker)
		}
```